extended/src/test/java/io/kubernetes/client/extended/leaderelection/LeaderElectionTest.java

```diff
@@ -285,16 +285,26 @@ public void testLeaderElectionCaptureException() throws ApiException, Interrupte
     leaderElectionConfig.setLeaseDuration(Duration.ofMillis(1000));
     leaderElectionConfig.setRetryPeriod(Duration.ofMillis(200));
     leaderElectionConfig.setRenewDeadline(Duration.ofMillis(700));
+    final Semaphore sem = new Semaphore(1);
     LeaderElector leaderElector =
-        new LeaderElector(leaderElectionConfig, (t) -> actualException.set(t));
+        new LeaderElector(
+            leaderElectionConfig,
+            (t) -> {
+              actualException.set(t);
+              sem.release();
+            });
 
     ExecutorService leaderElectionWorker = Executors.newFixedThreadPool(1);
+    sem.acquire();
     leaderElectionWorker.submit(
         () -> {
           leaderElector.run(() -> {}, () -> {});
         });
-    // TODO: Remove this sleep
-    Thread.sleep(Duration.ofSeconds(2).toMillis());
+    while (!sem.tryAcquire()) {
```

Comment:
Is there a reason for this loop? I would remove all this code and just do:

`sem.acquire()` like you did above.

Reply:
i wanted to print message on why we are waiting.. 
but i have changed to it to countdown latch, as i thought it would be better  than sempahore in these situations